controllers/managedocs_controller.go

```diff
@@ -522,6 +527,12 @@ func (r *ManagedOCSReconciler) reconcilePhases() (reconcile.Result, error) {
 				return ctrl.Result{}, err
 			}
 		}
+		// Reconciles only for provider deployment type
+		if strings.ToLower(r.DeploymentType) == "provider" {
+			if err := r.reconcileOnboardinValidationSecret(); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
```

Comment:
Can we move the check inside the `≈reconcileOnboardinValidationSecret` as an early exit?
So we have something like:

```
if !strings.ToLower(r.DeploymentType) != "provider" {
     r.Log.Info("Non provider deployment, skipping reconcile for onboardin validation secret")
     return nil
}
```